addons/web_editor/static/src/components/media_dialog/file_selector.xml

```diff
@@ -2,6 +2,7 @@
 <templates id="template" xml:space="preserve">
 <t t-name="web_editor.FileSelectorControlPanel" owl="1">
     <div class="d-flex flex-wrap gap-2 mt-4 mb-2 align-items-end">
+        <!-- TODO: remove the needle prop in master -->
         <SearchMedia searchPlaceholder="props.searchPlaceholder" needle="props.needle" search="props.search"/>
```

Comment:
as I said IRL, the removal of this needle made me think if it was possible to have a state where the FileSelector or IconSelector display results of one thing, but the input displays something else (since the two will not be synced).

And I believe it's possible in this case:
- debounce 1 (car) triggers RPC 1 which takes 5 seconds.
- debounce 2 (desks) triggers RPC 2 which take 1 second.
- RPC 2 finished and displays the result of desks
- RPC 1 which takes 5 seconds resolves, which then replaces the search results with car, while the input still displays desks

Though maybe I'm wrong, from what I've seen, debounce does not cancel / ignores promises. Instead, we should use [KeepLast](https://github.com/odoo/odoo/blob/7c9f883dc508a7a8a45bf7bf7e900da0be9b34be/addons/web/static/src/core/utils/concurrency.js#L7) as it seems to be used for other search fields which trigger RPC requests in Odoo.

Reply:
Added `KeepLast` mechanism and removed comments about removing the `needle` later because it is used on `loadMore` too. Now the `needle` always reflects the last `search`, be it completed or not.